app/controllers/application_controller.rb

```diff
@@ -3,7 +3,9 @@
 class ApplicationController < ActionController::Base
   # Returns the current signed in User (if any)
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id])
+    # Returns the extended session cookie value if present instead of the session cookie that expires on browser exit
+    user_id = cookies.encrypted[:_extended_session].presence || session[:user_id]
```

Comment:
```suggestion
    user_id = cookies.encrypted[:_extended_session] || session[:user_id]
```

Reply:
Besides, I don't think that having two sources of truth is good it causes confusing TBH and may raise issues.

I'm a strong fallower of the separation of concerns design principle [CHECK HERE](https://en.wikipedia.org/wiki/Separation_of_concerns).

When the user enables `remember_me`, the session cookie itself isn't enough to persist its data so we need another persistent placeholder or store.

So, the `_extended_session` concern/responsibility is to hold that data and provide it to the session when it needs it and the `session` cookie is the only one that should be the source of truth for the user data while the other is a simple store.

It's the same concept in computer software, we may store data on the disk (our `_extended_session`) to persist it but when we start the software it will load it first to the RAM (our `session`) that will use it afterwards as its source of truth.

Reply:
With being said we can do something like:
```
  # Returns the current signed in User (if any)
  def current_user
    # Returns the extended session cookie value if present instead of the session cookie that expires on browser exit
    session[:user_id] ||= cookies.encrypted[:_extended_session]
    @current_user ||= User.find_by(id: session[:user_id])
  end
```

Reply:
That's a very good use of the ||= operator. Thank you. I'm on it.

Reply:
`cookies.encrypted[:_greenlight_session]` is managed by Rails itself so it makes since that it will overwrite your changes.